js/ad_manager_vast.js

```diff
@@ -640,6 +884,52 @@ OO.Ads.manager(function(_, $) {
     };
 
     /**
+    * This method pings all the ad's error URLs with a specific error code if the error URL
+    * contains the macro, "[ERRORCODE]". If ad has a parent wrapper, then go up the chain and ping
+    * wrapper's error urls as well.
+    * @public
+    * @method Vast#trackError
+    * @param {number} code Error code
+    * @param {boolean} currentAdId Ad ID of current ad
+    */
+    this.trackError = _.bind(function trackErrorHelper(code, currentAdId) {
+      if (currentAdId && currentAdId in this.errorInfo) {
+        pingURLs(this.errorInfo[currentAdId].errorUrls);
+        var parentId = this.errorInfo[currentAdId].wrapperParentId;
+
+        // ping parent wrapper's error urls too if ad had parent
+        if (parentId) {
+          this.trackErrorHelper(code, parentId);
```

Comment:
Hmm, so I was initially advised to use a named function expression for recursive functions. Not entirely understand why, I found this on stackoverflow (http://stackoverflow.com/questions/7065120/calling-a-javascript-function-recursively). So I did it mainly for this, but since we don't really have that issue in our code, actually it doesn't make much sense to use a named function expression. I can remove it!